src/components/learnDetail/memo/MemoDotButton.tsx

```diff
@@ -0,0 +1,78 @@
+import ImageDiv from '@src/components/common/ImageDiv';
+import { icDotDefault, icDotHover } from 'public/assets/icons';
+import { useState, useRef, useEffect } from 'react';
+import styled from 'styled-components';
+import MemoPopup from './MemoPopup';
+import { IConfirmModalText } from '../ConfirmModal';
+import { IMemoHighlightId } from '@src/pages/learn/[id]';
+
+interface MemoDotButtonProps {
+  highlightId: number;
+  setMemoHighlightId: (id: IMemoHighlightId) => void;
+  setIsConfirmOpen: (open: boolean) => void;
+  setConfirmModalText: (text: IConfirmModalText) => void;
+}
+
+function MemoDotButton(props: MemoDotButtonProps) {
+  const { highlightId, setMemoHighlightId, setIsConfirmOpen, setConfirmModalText } = props;
+
+  const [isPopupOpen, setIsPopupOpen] = useState(false);
+  const memoPopupRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleClickOutside = (e: Event) => {
+      const eventTarget = e.target as HTMLElement;
+      if (isPopupOpen && !memoPopupRef?.current?.contains(eventTarget)) {
+        setIsPopupOpen(false);
+      }
+    };
+    window.addEventListener('click', handleClickOutside);
+    window.addEventListener('contextmenu', handleClickOutside);
+  }, [isPopupOpen]);
+
+  return (
+    <StMemoDotButton ref={memoPopupRef} onClick={() => setIsPopupOpen((prev) => !prev)}>
+      <StMemoDotImage isPopupOpen={isPopupOpen}>
+        <ImageDiv className="dot" src={icDotHover} alt="..." layout="fill" />
+        <ImageDiv className="dot default" src={icDotDefault} alt="..." layout="fill" />
+      </StMemoDotImage>
+      {isPopupOpen && (
+        <MemoPopup
+          highlightId={highlightId}
+          setMemoHighlightId={setMemoHighlightId}
+          setIsConfirmOpen={setIsConfirmOpen}
+          setConfirmModalText={setConfirmModalText}
+        />
+      )}
+    </StMemoDotButton>
+  );
+}
+
+export default MemoDotButton;
+
+const StMemoDotButton = styled.div`
+  position: absolute;
+  top: 1.2rem;
+  right: 1.2rem;
+
+  width: 4rem;
+  height: 4rem;
+
+  cursor: pointer;
+`;
+
+const StMemoDotImage = styled.div<{ isPopupOpen: boolean }>`
+  position: relative;
+
+  .dot {
+    position: absolute;
+    width: 4rem;
+    height: 4rem;
+    opacity: ${({ isPopupOpen }) => (isPopupOpen ? 1 : 0)};
```

Comment:
이거 popupOpen 일 때 dot 가 안보이게 조절하기 위한 코드맞나용??
투명도로 조절해줘도 되고 
```suggestion
    visibility: ${({ isPopupOpen }) => (isPopupOpen && "hidden"};
```
```suggestion
    display: ${({ isPopupOpen }) => (isPopupOpen && "none"};
```
요런식으로 아예 안보이는 효과를 주는 css 도 있어요!